### PR TITLE
Engine: Set data_type for all protobuf values

### LINKED ISSUE #45: Engine: Missing Data Type in Values returned when Getting Numeric Attributes
### Problem
When using these components types:
- `Meta` 
https://github.com/joeltio/bento-box/blob/b22420a2345004726071c54d3f0230b1052e2329/e2e/test_e2e.py#L25-L31
- `Velocity`
https://github.com/joeltio/bento-box/blob/b22420a2345004726071c54d3f0230b1052e2329/sdk/bento/example/specs.py#L19-L25

Retrieve attributes of type `int32`/`float64` will return a `Value` with `data_type` unset, causing the SDK to error due to unknown data type:
```
 TypeError: Unable to unwrap unsupported data type kind: None 
```

### Proposed Fix 
- [ ] Set `data_type` when in `Value` proto when retrieve numeric data types with `GetAttribute()` gRPC method.
- [ ] Add test to Engine to always check that `data_type` is always set in returned  `Value` protos to the SDK.

### PR DESCRIPTION
**Pull request by joeltio**
Fixes #45 

- Adds missing data types to all `Value`s.
- Updates all existing tests to ensure that data types are present for returned value
- Update `UserComponent` to also validate that values set have the correct `data_type`.
- Fix issue with "Something went wrong while running system with ID: " error message printing system index instead of system id.

### CODE REVIEW

**Review by mrzzy — COMMENTED** [review]
Looks mostly Ok. Give me sometime to test before merging.

**Comment by mrzzy** [comment]
If the user does not run step() the Simulation trying to `GetAttribute()` the data type is still missing. This is a good to fix.

**Comment by joeltio** [comment]
> If the user does not run step() the Simulation trying to `GetAttribute()` the data type is still missing. This is a good to have/fix.

This seems to be a problem because this branch was made before the #46 was merged in

File: sim/src/component/userComponent.cpp [hunk 1]
```diff
@@ -18,6 +19,20 @@ void UserComponent::setValue(const std::string& attrName,
...
+            "Missing data type when setting value for attribute " + attrName);
+    }
+
+    auto schemaType = compDef.schema().at(attrName);
+    auto differencer = google::protobuf::util::MessageDifferencer();
+    if (!differencer.Compare(schemaType, value.data_type())) {
+        throw std::runtime_error(
+            "Data type of given value does not match schema type for "
+            "attribute " +
+            attrName);
```
[the remark this answers is not in the document]
Comment (joeltio) [reply sim/src/component/userComponent.cpp]: Added better messages in f34cfef

**Comment by mrzzy** [comment]
Seems like the E2E check is failing because the E2E assumed implicit type conversion which has not yet been merged (PR #51). I have patched the E2E tests and merged into this PR to see if that fixes the issue.

**Review by mrzzy — APPROVED** [review]
:tada: LGTM

**Pull request merged**